packages/core/test/unit/query-generator/create-table-query.test.ts

```diff
@@ -0,0 +1,342 @@
+      mssql: `IF OBJECT_ID(N'[myTable]', 'U') IS NULL CREATE TABLE [myTable] ([data] BLOB(16M));`,
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "myTable" ("data" BLOB(16M)); END`,
+    });
+  });
+
+  it('produces a CREATE TABLE query with a normalized BLOB attribute', () => {
+    expectsql(queryGenerator.createTableQuery('myTable', { data: sequelize.normalizeDataType(DataTypes.BLOB).toSql() }), {
+      default: 'CREATE TABLE IF NOT EXISTS [myTable] ([data] BLOB);',
+      'mariadb mysql': 'CREATE TABLE IF NOT EXISTS `myTable` (`data` BLOB) ENGINE=InnoDB;',
+      postgres: 'CREATE TABLE IF NOT EXISTS "myTable" ("data" BYTEA);',
+      mssql: `IF OBJECT_ID(N'[myTable]', 'U') IS NULL CREATE TABLE [myTable] ([data] VARBINARY(MAX));`,
+      db2: 'CREATE TABLE IF NOT EXISTS "myTable" ("data" BLOB(1M));',
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "myTable" ("data" BLOB(1M)); END`,
+    });
+  });
+
+  it('produces a CREATE TABLE query with a normalized long BLOB attribute', () => {
+    expectsql(queryGenerator.createTableQuery('myTable', { data: sequelize.normalizeDataType(DataTypes.BLOB('long')).toSql() }), {
+      'mariadb mysql': 'CREATE TABLE IF NOT EXISTS `myTable` (`data` LONGBLOB) ENGINE=InnoDB;',
+      postgres: 'CREATE TABLE IF NOT EXISTS "myTable" ("data" BYTEA);',
+      mssql: `IF OBJECT_ID(N'[myTable]', 'U') IS NULL CREATE TABLE [myTable] ([data] VARBINARY(MAX));`,
+      sqlite: 'CREATE TABLE IF NOT EXISTS `myTable` (`data` BLOB);',
+      snowflake: 'CREATE TABLE IF NOT EXISTS "myTable" ("data" LONGBLOB);',
+      db2: 'CREATE TABLE IF NOT EXISTS "myTable" ("data" BLOB(2G));',
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "myTable" ("data" BLOB(2G)); END`,
+    });
+  });
+
+  it('produces a CREATE TABLE query with an ENUM and a VARCHAR attribute', () => {
+    expectsql(queryGenerator.createTableQuery('myTable', { title: 'ENUM("A", "B", "C")', name: 'VARCHAR(255)' }), {
+      'mariadb mysql': 'CREATE TABLE IF NOT EXISTS `myTable` (`title` ENUM("A", "B", "C"), `name` VARCHAR(255)) ENGINE=InnoDB;',
+      postgres: 'CREATE TABLE IF NOT EXISTS "myTable" ("title" "public"."enum_myTable_title", "name" VARCHAR(255));',
+      mssql: `IF OBJECT_ID(N'[myTable]', 'U') IS NULL CREATE TABLE [myTable] ([title] ENUM("A", "B", "C"), [name] VARCHAR(255));`,
+      sqlite: 'CREATE TABLE IF NOT EXISTS `myTable` (`title` ENUM("A", "B", "C"), `name` VARCHAR(255));',
+      'snowflake db2': 'CREATE TABLE IF NOT EXISTS "myTable" ("title" ENUM("A", "B", "C"), "name" VARCHAR(255));',
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "myTable" ("title" ENUM("A", "B", "C"), "name" VARCHAR(255)); END`,
+    });
+  });
+
+  it('produces a CREATE TABLE query with two VARCHAR attributes with specified engine', () => {
```

Comment:
the varchar part does not seem relevant. No need to mention it in the title, and we could just use a single varchar attribute